warehouse/query-core/src/test/java/datawave/test/JexlNodeAssert.java

```diff
@@ -0,0 +1,335 @@
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's image is equal to the given one.
+     * 
+     * @param image
+     *            the image
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasImage(String image) {
+        isNotNull();
+        if (!Objects.equals(actual.image, image)) {
+            failWithMessage("Expected image to be %s but was %s", image, actual.image);
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's value is null.
+     * 
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasNullValue() {
+        isNotNull();
+        if (actual.jjtGetValue() != null) {
+            failWithMessage("Expected value to be null, but was %s", actual.jjtGetValue());
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's value is not null.
+     * 
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasNonNullValue() {
+        isNotNull();
+        if (actual.jjtGetValue() == null) {
+            failWithMessage("Expected value to not be null");
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's value is equal to the given one.
+     * 
+     * @param value
+     *            the value
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasValue(Object value) {
+        isNotNull();
+        if (!Objects.equals(actual.jjtGetValue(), value)) {
+            failWithMessage("Expected value to be %s, but was %s", value, actual.jjtGetValue());
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's parent is null.
+     * 
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasNullParent() {
+        isNotNull();
+        if (actual.jjtGetParent() != null) {
+            failWithMessage("Expected parent to be null");
+        }
+        return this;
+    }
+    
+    /**
+     * Verifies that the actual node's parent is not null.
+     * 
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasNonNullParent() {
+        isNotNull();
+        if (actual.jjtGetParent() == null) {
+            failWithMessage("Expected parent to not be null, but was %s", actual.jjtGetParent());
+        }
+        return this;
+    }
+    
+    /**
+     * Returns a new {@link JexlNodeAssert} that will perform assertions on the actual node's parent.
+     * 
+     * @return the new {@link JexlNodeAssert} for the parent
+     */
+    public JexlNodeAssert parent() {
+        isNotNull();
+        return assertThat(actual.jjtGetParent());
+    }
+    
+    /**
+     * Verifies that the actual node has no children.
+     * 
+     * @return this {@link JexlNodeAssert}
+     */
+    public JexlNodeAssert hasNoChildren() {
+        isNotNull();
```

Comment:
This could be `return hasNumChildren(0)`. Just to save a bit of code.

Reply:
Good suggestion, I like it.